src/messages/RolesCommand.ts

```diff
@@ -0,0 +1,184 @@
+} from 'discord.js';
+import Common from '../common';
+import RoleChannelManager from '../roles/RoleChannelManager';
+import GuildSettings from '../settings/GuildSettings';
+import { GuildInfo, RolesField } from '../settings/SettingsDB';
+import Command from './Command';
+
+export default class RolesCommand extends Command {
+  deploy: ChatInputApplicationCommandData = {
+    name: 'roles',
+    description: 'Manage roles users can give themselves',
+    options: [
+      {
+        type: 1,
+        name: 'remove',
+        description: 'Remove a role',
+        options: [
+          {
+            type: 8,
+            name: 'role',
+            description: 'The role to remove',
+            required: true,
+          },
+        ],
+      },
+      {
+        type: 1,
+        name: 'edit',
+        description: 'Edit an existing role',
+        options: [
+          {
+            type: 8,
+            name: 'role',
+            description: 'The role to edit',
+            required: true,
+          },
+          {
+            type: 3,
+            name: 'emoji',
+            description: 'The new emoji to use',
+          },
+          {
+            type: 3,
+            name: 'description',
+            description: 'The new description to use',
+          },
+        ],
+      },
+      {
+        type: 1,
+        name: 'add',
+        description: 'Add a role',
+        options: [
+          {
+            type: 3,
+            name: 'role_name',
+            description: 'The name of the role to add',
+            required: true,
+            choices: [],
+          },
+          {
+            type: 3,
+            name: 'emoji',
+            description: 'The emoji to use for the role',
+            required: true,
+          },
+          {
+            type: 3,
+            name: 'description',
+            description: 'The description shown in the role channel',
+          },
+        ],
+      },
+    ],
+    defaultMemberPermissions: Permissions.FLAGS.MANAGE_GUILD,
+    dmPermission: false,
+  };
+
+  async handleCommand(
+    args: readonly CommandInteractionOption<CacheType>[],
+    interaction: CommandInteraction<CacheType>
+  ): Promise<void> {
+    if (interaction.guildId === null) {
+      interaction.reply({
+        content: 'You are only allowed to perform this command in a guild',
+        ephemeral: true,
+      });
+      return;
+    }
+
+    const info: GuildInfo = await GuildSettings.settings(interaction.guildId);
+
+    switch (args[0].name) {
+      case 'add':
+        const newRole: RolesField = {
+          name: this.getParam(args, 0),
+          emoji: this.getParam(args, 1),
+        };
+        if (args[0].options!.length === 3)
+          newRole.description = this.getParam(args, 2);
+        info.roles.push(newRole);
+        interaction.reply({
+          content: `The role ${newRole.name} has been added (note that the role will only be created when somebody assigns it to themself)`,
```

Comment:
```suggestion
          content: `The role ${newRole.name} has been added (note that the role will only be created when somebody assigns it to themselves)`,
```
(Src: https://www.deepl.com/write)